### PR TITLE
Change reconciler logic to replace kubernetes resources instead of patching them

### PR DESCRIPTION
**Pull request by m00g3n**
Change reconciler logic to replace kubernetes resources instead of patching them

See also: https://github.com/kyma-incubator/reconciler/issues/330

### CODE REVIEW

File: pkg/reconciler/kubernetes/kubeclient/kubeclient.go [hunk 1]
```diff
@@ -168,7 +159,8 @@ func (kube *KubeClient) ApplyWithNamespaceOverride(u *unstructured.Unstructured,
 		_ = info.Refresh(obj, true)
 	}
 
-	_, patchedObject, err := patcher.Patch(info.Object, modified, info.Namespace, info.Name)
+	replace := newReplace(helper)
+	patchedObject, err := replace(info.Object, info.Namespace, info.Name)
```
Comment (pbochynski) [inline pkg/reconciler/kubernetes/kubeclient/kubeclient.go:163]: patchedObject is not the best variable name now :)

**Review by pbochynski — DISMISSED** [review]

**Review by rJankowski93 — APPROVED** [review]

**Pull request merged**